### PR TITLE
Support ResponseErrors from balena-register-device

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: patch
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: src/lib/register-device.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import factory = require('balena-register-device');
 import * as Bluebird from 'bluebird';
 import { getRequestInstance } from './request';
 
+const { ApiError } = factory;
+export { ApiError };
```
Comment (Page-) [inline src/lib/register-device.ts:6]:
```suggestion
export const { ApiError } = factory;
```

File: typings/balena-register-device.d.ts [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
 // TODO: Upstream types to the repo
 declare module 'balena-register-device' {
+	import TypedError = require('typed-error');
+	import { Response } from 'request';
+
+	factory.ApiError = class ApiError extends TypedError {
+		response: Response;
+	};
+
 	function factory({
```
Comment (Page-) [inline typings/balena-register-device.d.ts:12]: It'd be better to put this avoid the `factory.ApiError =` just to avoid confusing people

Comment (CameronDiver) [reply typings/balena-register-device.d.ts:12]: I'm not sure what you mean by this?

Comment (Page-) [reply typings/balena-register-device.d.ts:8]: The first `avoid` was meant to be `above` - but you did it anyway so it's all cool

**Comment by CameronDiver** [comment]
@balena-ci I self-certify!

**Pull request merged**